versions/models.py

```diff
@@ -88,17 +100,15 @@ def next_version(self, object):
         if object.version_end_date == None:
             return object
         else:
-            try:
-                next = self.get(
-                    Q(identity=object.identity),
-                    Q(version_start_date=object.version_end_date))
-            except MultipleObjectsReturned as e:
-                raise MultipleObjectsReturned(
-                    "next_version couldn't uniquely identify the next version of object " + str(
-                        object.identity) + " to be returned\n" + str(e))
-            except ObjectDoesNotExist as e:
+            next = self.filter(
+                Q(identity=object.identity),
+                Q(version_start_date__gte=object.version_end_date)
+            ).order_by('version_start_date').first()
```

Comment:
Good question.

The postgresql util allows adding a partially unique index that enforces that no two current versions exist at the same time, but yes, even in postgresql if the objects were manually manipulated, two non-current versions could have an overlapping validity period.

Perhaps two objects should be fetched here, and an overlap check performed.  It still wouldn't prevent anyone from manually creating that situation of two overlapping objects; they would never notice a problem unless they called version_next or version_previous.  What do you think?

To actually prevent the possibility of creating that situation, for postgresql, it seems like it could make sense to provide a helper to create an exclusion constraint for objects with the same identity ( see http://www.postgresql.org/docs/9.3/static/rangetypes.html#RANGETYPES-CONSTRAINT).  This would only be usable if the btree_gist postgresql extension is installed (this requires superuser permissions to install).